README.md

```diff
@@ -134,17 +134,27 @@ When creating a new package, first determine which kind of package you are creat
 
 ### `apps/*`
 
-Applications with a UI.
+Each app has it's own configuration
 
 - [`cms`][cms] - Self-hosted [Directus][directus] headless CMS
 - [`web`][web] - Next.js customer-facing website
 - [`api`][api] - API abstracting communications with Algod etc
 
 ### `libs/*`
 
-Shared packages used by this monorepo.
+For performance and code organization reasons, the [https://nx.dev/structure/applications-and-libraries](Nx docs) recommend
+putting as much functionality as possible into libs, even if the code is only used in a single app. In Nx, a lib is more than just a directory under the `libs/` directory. Each lib must have an entry in the workspace.json file for the lib to build and import correctly.
 
-- [`schemas`][schemas] - Shared code and typings
+Linting will fail for any lib code that tries to import code from an app. This means that lib code should never access things like
+global configuration variables or environment variables. (eg. `Configuration`)
+
+If you wanted to create a new library at the path `libs/shared/utils`, you'd use the nx generator...
+`nx generate @nrwl/node:lib utils --directory shared`
```

Comment:
Is the intention to put **all** shared code in `@algomart/shared`? Any reason not to do multiple shared libraries such as `@algomart/algorand`, `@algomart/utilities`, `@algomart/payments` etc? Because it looks like the import paths will need to be distinct per "sub-library" anyway:

```json
  // from tsconfig.base.json
  "@algomart/shared/utils": ["libs/shared/utils/src/index.ts"],
```

Reply:
<img width="264" alt="Screen Shot 2022-03-22 at 1 38 43 PM" src="https://user-images.githubusercontent.com/3632506/159542042-e65011b0-0965-428c-a15e-01e0232cc002.png">

This is what it looks like in the client's implementation - I agree we can probably move a number of these out of the somewhat redundant "shared" folder

Reply:
I mimicked the directory structure recommended in the Nx docs.  It's also how nx example project is setup.
https://github.com/nrwl/nx-examples/tree/master/libs/shared

![image](https://user-images.githubusercontent.com/1264533/159547135-a5117d7a-4f8b-4fbe-b77f-558c440261f8.png)

![image](https://user-images.githubusercontent.com/1264533/159547368-c4b22439-782c-4761-8987-f0fca4faa80b.png)

Reply:
An example of something that would go in `libs/api` would be endpoint handlers that only live in api.... eg. the code that currently lives in `apps/api/modules`.

Reply:
@smonn I just reread your comment.  To clarify, the directory `libs/shared` isn't its own library.  If you look in workspaces.json you can see that each individual sub-directory is it's own library, which means that they can each get built and execute tests in parallel.

Reply:
That all makes sense. Happy to lean into the recommended Nx approach here.